<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [ ]:
# Membuat dataframes dari dictionary
import pandas as pd

data = {
    "name": ["john", "doe", "jane", "tim"],
    "age": [28, 25, 21, 24],
    "isMale": [True, True, False, True], 
    "badge": ["gold", "silver", "silver", "gold"], 
}

df  = pd.DataFrame(data)
df.head()

name  age  isMale   badge
0  john   28    True    gold
1   doe   25    True  silver
2  jane   21   False  silver
3   tim   24    True    gold

<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [13]:
# Membaca data dan memasukannya kedalam suatu variabel
from sys import exit
from os.path import exists

path = "diamonds-1.csv"
if (not exists(path)):
  exit(f"[Error] : file {path} not found, please upload the file first to the main directory")

df2 = pd.read_csv(path)
df2 

Unnamed: 0  carat        cut color  ... price     x     y     z
0               1   0.23      Ideal     E  ...   326  3.95  3.98  2.43
1               2   0.21    Premium     E  ...   326  3.89  3.84  2.31
2               3   0.23       Good     E  ...   327  4.05  4.07  2.31
3               4   0.29    Premium     I  ...   334  4.20  4.23  2.63
4               5   0.31       Good     J  ...   335  4.34  4.35  2.75
...           ...    ...        ...   ...  ...   ...   ...   ...   ...
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[53940 rows x 11 columns]

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab: 
Perbedaan Series dengan List
- Series hanya dapat menyimpan data yang homogen (satu tipe data), sedangkan pada tipe data List kita bisa menyimpan data yang heterogen (berbeda tipe data).
- List adalah built in data type pada python, sedangkan Series merupakan kelas dari pandas
- index pada list berupa nilai integer yang terurut(mulai dari 0 sampai n-1), Sedangkan pada series index bisa berupa nilai string yang unik    
- terdapat perbedaan hasil/definisi operasi + , -, *, / 

Perbedaan Series dengan Dictionary
- pada series kita bisa mengimplementasikan metode slicing sedangkan pada dictionary tidak bisa.
- Dictionary adalah built in data type pada python, sedangkan Series merupakan kelas dari pandas
- terdapat perbedaan hasil/definisi operasi + , -, *, / 

Perbedaan dataframe dan series
- series merupakan data satu dimensi sedangkan dataframe merupakan data 2-dimensi
- jika diibaratkan pada tabel, series merupakan kolom pada tabel tsb sedangkan dataframe adalah keseluruhan tabel

In [14]:
# code here
data = ["jhon", "doe", "jane", "tim"]
sr = pd.Series(data)
sr

0    jhon
1     doe
2    jane
3     tim
dtype: object

<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [15]:
# code here 
print("Beberapa data pertama:")
display(df2.head())

print("\n\nBeberapa data terakhir:")
display(df2.tail())

print("\n\nBaris pertama:")
display(df2.head(1))

print("\n\nBaris terakhir:")
display(df2.tail(1))

firstPrice = df2["price"][0]
lastColor = df2["color"][df2.shape[0]-1]
print(f"\n\nBaris pertama dari column price : {firstPrice}")
print(f"Baris terakhir dari column color : {lastColor}")


Beberapa data pertama:


Unnamed: 0  carat      cut color clarity  ...  table  price     x     y     z
0           1   0.23    Ideal     E     SI2  ...   55.0    326  3.95  3.98  2.43
1           2   0.21  Premium     E     SI1  ...   61.0    326  3.89  3.84  2.31
2           3   0.23     Good     E     VS1  ...   65.0    327  4.05  4.07  2.31
3           4   0.29  Premium     I     VS2  ...   58.0    334  4.20  4.23  2.63
4           5   0.31     Good     J     SI2  ...   58.0    335  4.34  4.35  2.75

[5 rows x 11 columns]



Beberapa data terakhir:


Unnamed: 0  carat        cut color  ... price     x     y     z
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[5 rows x 11 columns]



Baris pertama:


Unnamed: 0  carat    cut color clarity  ...  table  price     x     y     z
0           1   0.23  Ideal     E     SI2  ...   55.0    326  3.95  3.98  2.43

[1 rows x 11 columns]



Baris terakhir:


Unnamed: 0  carat    cut color clarity  ...  table  price     x     y     z
53939       53940   0.75  Ideal     D     SI2  ...   55.0   2757  5.83  5.87  3.64

[1 rows x 11 columns]



Baris pertama dari column price : 326
Baris terakhir dari column color : D


<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: 

Deskriptis statistik merupakan sebuah metode yang berhubungan dengan pengumpulan, pengolahan dan penyajian data, sehingga bisa memberi informasi yang berguna. 

Informasi yang diperoleh pada pengolahan data ini adalah :
- ukuran pemusatan data : mean, median, dan modus
- ukuran penyebaran data : jangkauan, simpangan, varian
- kecenderungan suatu gugus data 

Sedangkan data dapat disajikan dengan beberapa cara antara lain dengan menggunakan:
- bentuk grafis : diagram garis, batang, histogram, dsb
- bentuk numerik : Central Tredency, Dispersion, Fractile, Skewness, Pengukuran Keruncingan.


<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [16]:
# code here
description = df2.describe()
description

Unnamed: 0         carat  ...             y             z
count  53940.000000  53940.000000  ...  53940.000000  53940.000000
mean   26970.500000      0.797940  ...      5.734526      3.538734
std    15571.281097      0.474011  ...      1.142135      0.705699
min        1.000000      0.200000  ...      0.000000      0.000000
25%    13485.750000      0.400000  ...      4.720000      2.910000
50%    26970.500000      0.700000  ...      5.710000      3.530000
75%    40455.250000      1.040000  ...      6.540000      4.040000
max    53940.000000      5.010000  ...     58.900000     31.800000

[8 rows x 8 columns]

In [17]:
print("Berdasarkan tabel di atas diperoleh data sebagai berikut:")
data = [
  {"name":"price", "label":"rata-rata", "key":"mean"},
  {"name":"depth", "label":"standar deviasi", "key":"std"},
  {"name":"carat", "label":"maksimum", "key":"max"}
]

for obj in data:
  name = obj.get("name")
  label = obj.get("label")
  key = obj.get("key")
  value = round(description[name][key],6)
  print(f"- Nilai {label} dari kolom \"{name}\" adalah {value} ")


Berdasarkan tabel di atas diperoleh data sebagai berikut:
- Nilai rata-rata dari kolom "price" adalah 3932.799722 
- Nilai standar deviasi dari kolom "depth" adalah 1.432621 
- Nilai maksimum dari kolom "carat" adalah 5.01 


Jawab: 
Berdasarkan tabel di atas, diperoleh data sebagai berikut:
- Nilai rata-rata dari kolom "price" : 3932.799722
- Nilai standar deviasi dari kolom "depth" : 1.432621
- Nilai maksimum dari kolom "carat" : 5.010000